brokers/theia-plugin-broker/broker/broker_test.go

```diff
@@ -0,0 +1,384 @@
+	sPort := strconv.Itoa(port)
+	endpointName := "port" + sPort
+	var re = regexp.MustCompile(`[^a-z_0-9]+`)
+	prettyID := re.ReplaceAllString(publisher+"_"+pubName, `_`)
+	expectedPlugins := []model.ChePlugin{
+		{
+			ID:      meta.ID,
+			Version: meta.Version,
+			Endpoints: []model.Endpoint{
+				{
+					Name:       endpointName,
+					Public:     false,
+					TargetPort: port,
+				},
+			},
+			Containers: []model.Container{
+				{
+					Name:  cname,
+					Image: image,
+					Volumes: []model.Volume{
+						{
+							Name:      "projects",
+							MountPath: "/projects",
+						},
+						{
+							Name:      "plugins",
+							MountPath: "/plugins",
+						},
+					},
+					Ports: []model.ExposedPort{
+						{
+							ExposedPort: port,
+						},
+					},
+					Env: []model.EnvVar{
+						{
+							Name:  "THEIA_PLUGIN_ENDPOINT_PORT",
+							Value: sPort,
+						},
+					},
+				},
+			},
+			WorkspaceEnv: []model.EnvVar{
+				{
+					Name:  "THEIA_PLUGIN_REMOTE_ENDPOINT_" + prettyID,
+					Value: "ws://" + endpointName + ":" + sPort,
+				},
+			},
+		},
+	}
+	return expectedPlugins
+}
+
+func Test_process_regular_plugin(t *testing.T) {
+	mockBroker = &TheiaPluginBroker{
+		bMock,
+		uMock,
+		storage.New(),
+	}
+	workDir := tests.CreateTestWorkDir()
+	defer tests.RemoveAll(workDir)
+	archivePath := filepath.Join(workDir, "pluginArchive")
+	unarchivedPath := filepath.Join(workDir, "plugin")
+	packageJSONPath := filepath.Join(unarchivedPath, "package.json")
+	meta := model.PluginMeta{
+		ID:      "test-id",
+		Version: "test-v",
+		URL:     "http://test.url",
+	}
+	pluginPath := filepath.Join("/plugins", fmt.Sprintf("%s.%s.theia", meta.ID, meta.Version))
+	packageJSON := packageJson{
+		Name:      "test-name",
+		Publisher: "test-publisher",
+		Engines: engines{
+			CheRuntimeContainer: "",
+		},
+	}
+	expectedPlugins := []model.ChePlugin{
+		{
+			ID:      meta.ID,
+			Version: meta.Version,
+		},
+	}
+	bMock.On("PrintDebug", mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string"))
+	bMock.On("PrintDebug", mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string"), mock.AnythingOfType("string"))
+	uMock.On("TempDir", "", "theia-plugin-broker").Return(workDir, nil).Once()
+	uMock.On("Download", "http://test.url", archivePath).Return(nil).Once()
+	uMock.On("Unzip", archivePath, unarchivedPath).Once().Return(func(archive string, dest string) error {
+		tests.CreateDirByPath(dest)
+		tests.CreateFileWithContent(packageJSONPath, tests.ToJSONQuiet(packageJSON))
+		return nil
+	}).Once()
+	uMock.On("CopyFile", archivePath, pluginPath).Return(nil).Once()
+
+	err := mockBroker.processPlugin(meta)
+
+	assert.Nil(t, err)
+	plugins, err := mockBroker.storage.Plugins()
+	assert.Nil(t, err)
+	assert.Equal(t, expectedPlugins, *plugins)
+	bMock.AssertExpectations(t)
```

Comment:
`bMock`, `uMock` looks unclear for me here. Maybe it would be better to name it just `broker`, `ioUtil` or `brokerMock`, `ioUtilMock`. Or it is a common practise in `go` to name variables in tests in this way (`bMock`, `uMock`)?

Reply:
AFAIK it is common for go to shorten names